### PR TITLE
Creature: add {Serialize|Deserialize}Quickbar()

### PR DESCRIPTION
**Pull request by Daztek**

### CODE REVIEW

**Comment by Daztek** [comment]
Moved from Player to Creature

File: Plugins/Creature/Creature.cpp [hunk 1]
```diff
@@ -1902,5 +1907,79 @@ ArgumentStack Creature::GetFlatFooted(ArgumentStack&& args)
...
+        uint8_t *pData = nullptr;
+        int32_t dataLength = 0;
+
+        CResGFF    resGff;
+        CResStruct resStruct{};
+
+        if (resGff.CreateGFFFile(&resStruct, "GFF ", "V2.0"))
+        {
+            pCreature->SaveQuickButtons(&resGff, &resStruct);
+            resGff.WriteGFFToPointer((void**)&pData, /*ref*/dataLength);
```
Comment (mtijanic) [inline Plugins/Creature/Creature.cpp:1925]: Yep, this leaks `pData`. Needs a `delete [] pData;` after encoding. Same problem in Serialize.cpp

Comment (Daztek) [reply Plugins/Creature/Creature.cpp:1925]: Fixed

File: NWNXLib/Serialize.cpp [hunk 2]
```diff
@@ -85,7 +85,10 @@ std::vector<uint8_t> SerializeGameObject(CGameObject *pObject, bool bStripPCFlag
             break;
     }
 
-    return std::vector<uint8_t>(pData, pData+dataLength);
+    std::vector<uint8_t> serialized(pData, pData+dataLength);
+    delete[] pData;
+
+    return serialized;
```
Comment (mtijanic) [inline NWNXLib/Serialize.cpp:91]: return `std::move(serialized)`

**Pull request merged**